build-version.sh

```diff
@@ -30,6 +30,20 @@ if [ "$TARGET_ARCH" == "x86_64-unknown-linux-musl" ]; then
 
     CARGO_BIN_DIR="${TEMPDIR}/cargo/bin"
     CRATES2_JSON_PATH="${TEMPDIR}/cargo/.crates2.json"
+elif [ "$TARGET_ARCH" == "aarch64-unknown-linux-gnu" ]; then
+    mkdir -p zigfolder
+    curl "$(curl -q https://ziglang.org/download/index.json | jq 'to_entries | map([.key, .value])[1][1]["x86_64-linux"] | .tarball' | sed -e 's/^"//' -e 's/"$//')" | tar -xJ -C zigfolder --strip-components 1
+
+    export PATH="$PWD/zigfolder:$PATH"
+    rustup target add "$TARGET_ARCH"
+    if ! [ -f "$HOME/.cargo/config" ]; then
+        echo "[target.aarch64-unknown-linux-gnu]" >>~/.cargo/config
+        echo 'linker = "/home/runner/work/cargo-quickinstall/cargo-quickinstall/zig-aarch64.sh"' >>~/.cargo/config
```

Comment:
I managed to make https://github.com/cargo-bins/cargo-quickinstall/actions/runs/4034113715/jobs/6935051984 pass. I think the thing that did the trick was this:

```suggestion
        echo "linker = \"$PWD/zig-aarch64.sh\"" >>~/.cargo/config
```

You can also try cherry-picking the thing that makes our selftest stuff checkout the repo in the same place as the builder does. Annoyingly it is /home/runner/work/cargo-quickinstall/cargo-quickinstall/cargo-quickinstall/ :( Might be a thing for another PR though.